src/main/java/domain/AttendanceDate.java

```diff
@@ -0,0 +1,56 @@
+package domain;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.MonthDay;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.util.Objects;
+
+public class AttendanceDate {
+
+    private final LocalDate date;
+
+    public AttendanceDate(LocalDate date) {
+        validateWeekendAndHoliday(date);
+        this.date = date;
+    }
+
+    private static void validateWeekendAndHoliday(LocalDate date) {
+        if (date.getDayOfWeek() == DayOfWeek.SATURDAY || date.getDayOfWeek() == DayOfWeek.SUNDAY || isHoliday(date)) {
```

Comment:
`isHoliday()`와 `date.getDayOfWeek() == DayOfWeek.SATURDAY || date.getDayOfWeek() == DayOfWeek.SUNDAY`가 추상화 수준이 안 맞는 것 같습니다. 아래와 같이 수정해보면 어떨까요?

```suggestion
        if (isWeekend(date) || isHoliday(date)) {
```